### DISCUSSION TITLE
Simpleadmin remote upgrade

### DISCUSSION THREAD

**Discussion by nonameirl** [iamromulan/cellular-modem-wiki#60] [Help - SDXLEMUR/SDXPRAIRIE - Quectel]
I am planning to upgrade Simple Admin to version 2.0.
Since I manage the device remotely, I want to ensure I follow the correct steps before proceeding.

I attempted via RMxxx_rgmii_toolkit.sh using option 2 and this was the result:

```
Installing Simpleadmin 2.0
Entware/OPKG is already installed.
useradd already exists. Continuing...
curl already exists. Continuing...
2) Installing Simpleadmin 2.0
/usrdata/simpleupdates/scripts/update_socat-at-bridge.sh: No space left on device
Installing/updating dependency: socat-at-bridge
Please Wait....
./RMxxx_rgmii_toolkit.sh: line 334: /usrdata/simpleupdates/scripts/update_socat-at-bridge.sh: not found
 Dependency: socat-at-bridge has been updated/installed.
/usrdata/simpleupdates/scripts/update_simplefirewall.sh: No space left on device
Installing/updating dependency: simplefirewall
Please Wait....
./RMxxx_rgmii_toolkit.sh: line 340: /usrdata/simpleupdates/scripts/update_simplefirewall.sh: not found
 Dependency: simplefirewall has been updated/installed.
Entware/OPKG is already installed.
useradd already exists. Continuing...
curl already exists. Continuing...
Collected errors:
 * opkg_conf_load: Could not create lock file /opt/tmp/opkg.lock: No space left on device.
Collected errors:
 * opkg_conf_load: Could not create lock file /opt/tmp/opkg.lock: No space left on device.
/usrdata/root/bin/htpasswd: No space left on device
/usrdata/root/bin/simplepasswd: No space left on device
To change your simpleadmin (admin) password in the future...
In the console type simplepasswd and press enter
./RMxxx_rgmii_toolkit.sh: line 315: /usrdata/root/bin/simplepasswd: not found
/usrdata/simpleupdates/scripts/update_simpleadmin.sh: No space left on device
Installing/updating: Simpleadmin content
Please Wait....
./RMxxx_rgmii_toolkit.sh: line 347: /usrdata/simpleupdates/scripts/update_simpleadmin.sh: not found
Simpleadmin content has been updated/installed.
```

**Comment by MickMorley** [comment]
Here are the steps that I took to upgrade my Simple Admin instance. 

First I checked how old my deployment was, it was setup in December via running `ls -l /usrdata/simpleadmin/`

![image](https://github.com/user-attachments/assets/31c2c3a2-aea5-4e2d-944b-a3f83bd24c31)

Then you can type in `menu` to bring up the Simple Console Menu and hit `3` to enter the Toolkit

Hit `1` to download the Toolkit and you will be presented with the Toolkit options

![image](https://github.com/user-attachments/assets/13a5dd15-464c-4882-b670-2b5ca360f6d5)

Once you hit `2` it will install/update the latest toolkit.

You will be asked to update your `root` password and then it will disconnect. Give it a few minutes before you try to reconnect so it will finish installing.

If you want to make any tweaks via the Toolkit, you can repeat the steps above to get back to the toolkit.

![image](https://github.com/user-attachments/assets/b2d9f45c-d7e3-4daf-bb7d-fa4a443ea3bb)

Notes: I upgraded to for the fix https://github.com/iamromulan/cellular-modem-wiki/discussions/100

Edit: I just realized you are going from Simple Admin to Simple Admin 2.0.  My process was for Simple Admin 2.0 to the latest Simple Admin 2.0.  My guess is you need to physically connect to it to upgrade to 2.0

**Comment by clndwhr** [comment]
@nonameirl  /@MickMorley, I believe this is a combination of both going from SA to SA2 AND during the install process the device did not have enough space on the device to be able to execute the installation process as indicated by the: `/usrdata/simpleupdates/scripts/update_socat-at-bridge.sh: No space left on device` error
